cmd/krew/cmd/install.go

```diff
@@ -90,45 +96,56 @@ Remarks:
 				return errors.New("--archive can be specified only with --manifest or --manifest-url")
 			}
 
-			var install []index.Plugin
+			var install []pluginEntry
 			for _, name := range pluginNames {
-				plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(constants.DefaultIndexName), name)
+				indexName, pluginName := pathutil.CanonicalPluginName(name)
+				plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(indexName), pluginName)
 				if err != nil {
 					if os.IsNotExist(err) {
 						return errors.Errorf("plugin %q does not exist in the plugin index", name)
 					}
 					return errors.Wrapf(err, "failed to load plugin %q from the index", name)
 				}
-				install = append(install, plugin)
+				install = append(install, pluginEntry{
+					p:         plugin,
+					indexName: indexName,
+				})
 			}
 
 			if *manifest != "" {
 				plugin, err := indexscanner.ReadPluginFromFile(*manifest)
 				if err != nil {
 					return errors.Wrap(err, "failed to load plugin manifest from file")
 				}
-				install = append(install, plugin)
+				install = append(install, pluginEntry{
+					p:         plugin,
+					indexName: constants.DefaultIndexName,
```

Comment:
this is a case we did not think of.
I guess we can keep shoving them into default. thoughts @corneliusweig ?

Reply:
This is what happens at the moment. But we can use the opportunity to separate this better from the `default` index. So instead, what about setting the indexName to `detached` or something similar?

Reply:
++ @chriskim06 do you mind making a small PR?